### PR TITLE
[Backport v3.0-branch] Armv8r misc fix

### PR DESCRIPTION
**Pull request by povergoing**
Backport 4 commits from #43845:

Mainly fix issues caused by FVP_BaseR_AEMv8R >= 11.17
```
- Improve the FVP version check based on the suggestions from @tejlmand. Print a warning message on 'west build -t run', if the FVP does not meet the minimal requirement. But I think it's better not to stop executing the FVP program.
- Remove the parameters of FVP. The FVP_BaseR_AEMv8R >= 11.17 has changed its parameters '-C bp.dram.enable_atomic_ops' to '-C bp.s_dram.enable_atomic_ops' and '-C bp.sram.enable_atomic_ops' to '-C bp.s_sram.enable_atomic_ops', which is very annoying. I think the feature is unnecessary, so remove it. At the same time, we should disable the LSE feature of GCC with Armv8.4-a.
- Enable AARCH64_IMAGE_HEADER by default for Armv8R AArch64
- Fix the SRAM/FLASH to the right address. The previous address is wrong because I remember FVP has an issue with the FLASH address at some of the beta versions and uses DRAM to simulate FLASH. Now it's time to fix it.
```
fix auto backport fail #45451

### CODE REVIEW

**Review by stephanosio — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]

**Comment by dkalowsk** [comment]
@povergoing can you summarize these (and your other PR) changes into the release-notes as well?

**Comment by povergoing** [comment]
> @povergoing can you summarize these (and your other PR) changes into the release-notes as well?

Sure, no problem. But I haven't done that, and have no idea where the `release-notes` is, do you have brief docs/instructions about how-to?

**Comment by dkalowsk** [comment]
@povergoing On the v3.0-branch add to the doc/releases/release-notes-3.0.rst after https://github.com/zephyrproject-rtos/zephyr/pull/45606 is merged in.

**Comment by povergoing** [comment]
> @povergoing On the v3.0-branch add to the doc/releases/release-notes-3.0.rst after #45606 is merged in.

Got it
